src/libcore/macros.rs

```diff
@@ -519,54 +519,69 @@ macro_rules! unreachable {
     });
 }
 
-/// Indicates unfinished code.
+/// Indicates unfinished code by panicking with a message of "not yet implemented".
 ///
-/// This can be useful if you are prototyping and are just looking to have your
-/// code type-check, or if you're implementing a trait that requires multiple
-/// methods, and you're only planning on using one of them.
+/// This is useful if you are prototyping and want your program to type-check
+/// and stop execution whenever it reaches logic that has not yet been implemented.
```

Comment:
I removed that because I thought "implementing a trait but only caring about some methods" would fall under the more generic "want your program to type-check" explanation.

However, I think it does make sense that it would be a very common use care for this that would make the macro use more clear. I'll add this back.